libs/@local/codegen/src/definitions/type.rs

```diff
@@ -0,0 +1,60 @@
+use alloc::borrow::Cow;
+
+use specta::datatype;
+
+use super::{Enum, Primitive};
+
+#[derive(Debug, Clone)]
+pub enum Type {
+    Reference(Cow<'static, str>),
```

Comment:
can we guarantee that the name is unique here? What about cases of:

```
struct X { .. }

fn foo() {
	struct X { .. }
}
```

I believe that these will have the same name, so there'd be some ambiguity here in resolving the type.

I think it'd make sense to instead rely on our own `TypeId` that's just a monotonically increasing counter and maps to a SID/Facet ID.

Reply:
Yes, that's going to be done in a follow up. The corresponding ticket is H-4486. For now I will just wrap `SID` and this can be replaced later by the facet ID.